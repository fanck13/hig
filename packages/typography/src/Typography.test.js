import React from "react";
import { shallow } from "enzyme";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import Typography from "./Typography";

describe("Typography", () => {
  function newProps(variant, ...args) {
    const variantDetails = variant ? `${variant} variant` : "Default";
    const otherDetails = args.length ? ` with ${args.join(" ")}` : "";
    return {
      variant,
      children: `${variantDetails} should render nicely${otherDetails}.`
    };
  }

  it("renders", () => {
    expect(
      shallow(<Typography>It should render nicely</Typography>)
    ).toMatchObject(/It should render nicely/);
  });

  takeSnapshotsOf(Typography, [
    // new API tests
    { description: "renders default Typography", props: newProps() },
    {
      description: "renders Typography with align and fontWeight props",
      props: Object.assign(
        newProps("body", "align center and fontWeight bold"),
        { align: "center", fontWeight: "bold" }
      )
    },
    {
      description: "renders body variant Typography",
      props: newProps("body")
    },
    {
      description: "renders caption variant Typography",
      props: newProps("caption")
    },
    {
      description: "renders h1 variant Typography",
      props: newProps("h1")
    },
    {
      description: "renders h2 variant Typography",
      props: newProps("h2")
    },
    {
      description: "renders h3 variant Typography",
      props: newProps("h3")
    }
  ]);
});
