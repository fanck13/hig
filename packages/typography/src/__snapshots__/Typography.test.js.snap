// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography  1`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  Default should render nicely.
</span>
`;

exports[`Typography  2`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: bold;
  line-height: 1.3;
  margin: 0;
  text-align: center;
}

<span
  className="emotion-0"
>
  body variant should render nicely with align center and fontWeight bold.
</span>
`;

exports[`Typography  3`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  body variant should render nicely.
</span>
`;

exports[`Typography  4`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 12px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  caption variant should render nicely.
</span>
`;

exports[`Typography  5`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  h1 variant should render nicely.
</span>
`;

exports[`Typography  6`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 24px;
  font-weight: 400;
  line-height: 1.3;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  h2 variant should render nicely.
</span>
`;

exports[`Typography  7`] = `
.emotion-0 {
  color: #3c3c3c;
  display: block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 20px;
  font-weight: 600;
  line-height: 1.4;
  margin: 0;
  text-align: initial;
}

<span
  className="emotion-0"
>
  h3 variant should render nicely.
</span>
`;
